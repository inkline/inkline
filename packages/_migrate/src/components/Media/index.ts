export { default as Media } from '@inkline/inkline/components/Media/Media.vue';
