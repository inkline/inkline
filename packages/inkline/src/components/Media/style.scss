@import '../../css/config/index';
@import '../../css/mixins/index';
@import 'css/_variables';

/**
 * Media
 */
.media {
    position: relative;
    display: flex;
    flex-direction: row;

    > .body {
        flex: 0 1 auto;

        > *:first-child {
            margin-top: 0;
        }

        > *:last-child {
            margin-bottom: 0;
        }
    }

    > img,
    > .img,
    > .image {
        align-self: flex-start;
        height: auto;
        flex: 0 0 auto;
        margin-right: $media-image-margin-right;
    }
}
