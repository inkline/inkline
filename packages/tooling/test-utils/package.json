{
    "name": "@inkline/test-utils",
    "version": "1.0.0",
    "type": "module",
    "scripts": {
        "dev": "devtools tsup watch",
        "build": "devtools tsup build",
        "lint": "devtools eslint"
    },
    "dependencies": {
        "@inkline/vue": "workspace:*"
    },
    "devDependencies": {
        "@inkline/devtools": "workspace:*",
        "@inkline/eslint-config": "workspace:*",
        "@inkline/types": "workspace:*",
        "@types/node": "catalog:",
        "@types/shelljs": "catalog:",
        "eslint": "catalog:",
        "tsx": "catalog:",
        "tsup": "catalog:",
        "typescript": "catalog:",
        "type-fest": "catalog:",
        "vue": "catalog:"
    },
    "peerDependencies": {
        "vue": ">=3.0.0"
    },
    "files": [
        "dist",
        "LICENSE"
    ],
    "types": "./dist/index.d.ts",
    "main": "./dist/index.js",
    "module": "./dist/index.cjs",
    "exports": {
        ".": {
            "types": "./dist/index.d.ts",
            "import": "./dist/index.js",
            "require": "./dist/index.cjs"
        },
        "./*": "./*"
    },
    "funding": {
        "type": "github",
        "url": "https://github.com/sponsors/alexgrozav"
    },
    "bugs": {
        "url": "https://github.com/inkline/inkline/issues"
    },
    "homepage": "https://inkline.io",
    "keywords": [
        "inkline",
        "ui",
        "ux",
        "dx",
        "library",
        "framework",
        "typescript"
    ],
    "engines": {
        "node": "*",
        "pnpm": "*"
    },
    "license": "ISC",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/inkline/inkline.git"
    }
}
