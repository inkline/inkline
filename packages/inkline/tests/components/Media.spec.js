import { shallowMount } from '@vue/test-utils';
import Media from '@inkline/inkline/src/components/Media';

describe('Components', () => {
    describe('Media', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = shallowMount(Media);
        });

        it('should be named correctly', () => {
            expect(Media.name).toEqual('IMedia');
        });

        it('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
