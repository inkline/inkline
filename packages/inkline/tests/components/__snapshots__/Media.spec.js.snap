// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Media should render correctly 1`] = `
<div class="media">
  <div class="body"></div>
</div>
`;
